src/test/java/org/broadinstitute/consent/http/db/MailMessageDAOTest.java

```diff
@@ -207,4 +212,105 @@ public void testInsert_MissingCreateDate() {
                 null
         );
     }
+
+    @Test
+    public void testFetch() {
+        EnumSet.allOf(EmailType.class).forEach(t -> {
+            Instant now = Instant.now();
+            mailMessageDAO.insert(
+                    RandomStringUtils.randomAlphanumeric(10),
+                    RandomUtils.nextInt(1, 1000),
+                    RandomUtils.nextInt(1, 1000),
+                    t.getTypeInt(),
+                    now,
+                    RandomStringUtils.randomAlphanumeric(10),
+                    RandomStringUtils.randomAlphanumeric(10),
+                    RandomUtils.nextInt(200, 399),
+                    now
+            );
+        });
+
+        EnumSet.allOf(EmailType.class).forEach(t ->
+                assertEquals(1, mailMessageDAO.fetchMessagesByType(t.getTypeInt(), 1, 0).size()));
+    }
+
+    @Test
+    public void testFetchLimitAndOffset() {
+        Instant now = Instant.now();
+        mailMessageDAO.insert(
+                RandomStringUtils.randomAlphanumeric(10),
+                RandomUtils.nextInt(1, 1000),
+                RandomUtils.nextInt(1, 1000),
+                EmailType.COLLECT.getTypeInt(),
+                now,
+                RandomStringUtils.randomAlphanumeric(10),
+                RandomStringUtils.randomAlphanumeric(10),
+                RandomUtils.nextInt(200, 399),
+                now
+        );
+
+        List<MailMessage> mailMessageList = mailMessageDAO.fetchMessagesByType(EmailType.COLLECT.getTypeInt(), 1, 0);
+        assertEquals(1, mailMessageList.size());
+
+        List<MailMessage> mailMessageList2 = mailMessageDAO.fetchMessagesByType(EmailType.COLLECT.getTypeInt(), 1, 1);
+        assertEquals(0, mailMessageList2.size());
+
+        Integer mailId2 = mailMessageDAO.insert(
+                RandomStringUtils.randomAlphanumeric(10),
+                RandomUtils.nextInt(1, 1000),
+                RandomUtils.nextInt(1, 1000),
+                EmailType.COLLECT.getTypeInt(),
+                now,
+                RandomStringUtils.randomAlphanumeric(10),
+                RandomStringUtils.randomAlphanumeric(10),
+                null,
+                now
+        );
+
+        List<MailMessage> mailMessageList3 = mailMessageDAO.fetchMessagesByType(EmailType.COLLECT.getTypeInt(), 1, 1);
+        assertEquals(1, mailMessageList3.size());
+        assertEquals(mailId2, mailMessageList3.get(0).getEmailId());
+
+        List<MailMessage> mailMessageList4 = mailMessageDAO.fetchMessagesByType(EmailType.COLLECT.getTypeInt(), 20, 0);
+        assertEquals(2, mailMessageList4.size());
+    }
+
+    @Test
+    public void testFetchByCreateDate_with_limit_and_offset(){
+        Date start = new Date();
+        generateMessagesInTime(2, 1);
+        Date end = new Date();
```

Comment:
could using the current time via `new Date()` lead to any sort of flakiness? maybe we should have more fixed dates in the test. also, I am having trouble understanding exactly what this test does / if it is complete, maybe splitting into a couple smaller tests would be good. also having the test verify the exact messages it gets instead of just the size could be good too

Reply:
I am similarly confused by this test. I get the general idea that we want to find messages between dates with various offsets and limits - each of the 4 cases need some commentary on what they should be finding. I think what this is also testing is `BETWEEN SYMMETRIC`, but I can't be sure. The confusing part to me is start and end dates that are effectively just milliseconds apart.

Here's my understanding
---
Given: 
* A `start` date of `now`
* A list of 2 mail messages. First message is created with an instant of `now` + a few milliseconds. Second message is created with the same instant, but one day earlier.
* An `end` date that's a few milliseconds after the first message creation date.
* A `startCalendar` which represents a 3rd search date that should be a few milliseconds after `end`

Case 1, `mailMessageList`: This makes sense - this should return the first mail message since it is the only one between start and end

Case 2, `mailMessageList2`: This doesn't make sense to me. It's expecting 2 results, but `startCalendar` and `end` are just a few milliseconds apart. This should not be finding message #2

Case 3, `mailMessageList3`: This also doesn't make sense to me for the same reason. The offset makes sense - if you're expecting 2 messages, and offsetting it by 1, the list should be 1 element long. Since the date range is just a few milliseconds apart, and both are after **_start_**, it shouldn't find any of the messages.

Case 4, `mailMessageList4`: Same here, searching between two dates that are both after the creation dates of both messages should return 0 messages.